### PR TITLE
New Dispatcher implementation for Academic Calendar

### LINKED ISSUE #239: Implement Academic Calendar Dispatcher

### PR DESCRIPTION
**Pull request by rjcanto**
Corrected downloader tasklet.

Closes #239

### CODE REVIEW

File: src/main/kotlin/org/ionproject/integration/job/ISELAcademicCalendarJob.kt [hunk 1]
```diff
@@ -33,25 +35,38 @@ class ISELAcademicCalendarJob(
...
+    private fun taskletStep(name: String, tasklet: Tasklet): TaskletStep {
+        return stepBuilderFactory
+            .get(name)
+            .tasklet(tasklet)
+            .build()
+    }
+
+    @StepScope
+    @Bean
+    fun downloadCalendarPDFAlternateTasklet(): DownloadAndCompareTasklet {
```
Comment (rjcanto) [inline src/main/kotlin/org/ionproject/integration/job/ISELAcademicCalendarJob.kt:63]: to be reviewed in #249

File: src/main/kotlin/org/ionproject/integration/dispatcher/ISELAcademicCalendarDispatcherImpl.kt [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+            name = props.gitRepository,
+            url = props.gitRepoUrl,
+            user = props.gitUser,
+            password = props.gitPassword,
+            branch = props.gitBranchName
+        )
+        gitFactory.checkout(props.stagingFilesDir.path, data, props.timeoutInSeconds)
+    }
+
+    override fun dispatch(data: AcademicCalendarData, format: OutputFormat): DispatchResult =
```
Comment (rjcanto) [inline src/main/kotlin/org/ionproject/integration/dispatcher/ISELAcademicCalendarDispatcherImpl.kt:39]: To be reviewed in #247

File: src/main/kotlin/org/ionproject/integration/job/ISELAcademicCalendarJob.kt [hunk 3]
```diff
@@ -33,25 +35,38 @@ class ISELAcademicCalendarJob(
...
     val downloader: IFileDownloader,
+    val dispatcher: IAcademicCalendarDispatcher,
     @Autowired
     val ds: DataSource
 ) {
 
-    @Value("src/test/resources/calendarTest.pdf")
-    lateinit var inputPdf: String
-    lateinit var dispatcher: IAcademicCalendarDispatcher
+    // lateinit var dispatcher: IAcademicCalendarDispatcher
```
Comment (grimord) [inline src/main/kotlin/org/ionproject/integration/job/ISELAcademicCalendarJob.kt]: Forgotten? Probably should be removed

Comment (rjcanto) [reply src/main/kotlin/org/ionproject/integration/job/ISELAcademicCalendarJob.kt]: indeed 😴

**Review by grimord — APPROVED** [review]

**Pull request merged**